tutormfe/plugin.py

```diff
@@ -86,6 +74,27 @@ def _add_core_mfe_apps(apps: dict[str, MFE_ATTRS_TYPE]) -> dict[str, MFE_ATTRS_T
     return apps
 
 
+@hooks.Actions.PROJECT_ROOT_READY.add()
+def _modify_mfes(root: str):
+    """
+    Set the GitHub ref for OpenEdX MFEs that are using the *.master branch.
+
+    If the repository of the MFE is not of openedx or the branch is not a
+    *.master branch, we are not setting the ref, hence the build won't benefit
+    from build caches.
+    """
+    config = tutor_config.load_minimal(root)
+
+    for mfe, app in iter_mfes():
+        mfe_app_version = app.get("version", config.get("MFE_COMMON_VERSION"))
+        is_openedx_repo = app["repository"].startswith("https://github.com/openedx/")
+        is_master_branch = re.match(r"(^open-release\/\w+\.)?master$", mfe_app_version) is not None
```

Comment:
Caret `^` should be at the beginning of the regexp, right?

    r"^(open-release\/\w+\.)?master$"

Reply:
Oh, what a typo. Thanks, I intended to have it at the beginning.